### PR TITLE
Handles an exception replicating 0-byte objects

### PR DESCRIPTION
**Pull request by idlira**
It is possible to upload 0-byte files to S3, but when using a multipart upload, an AmazonS3Exception will be thrown attempting to complete the upload if no parts are provided.

We detect this condition and ignore the error, as backing up a file with no content is pointless.

Fixes: [SIRI-651](https://scireum.myjetbrains.com/youtrack/issue/SIRI-651)

### CODE REVIEW

File: src/main/java/sirius/biz/storage/s3/ObjectStore.java [hunk 1]
```diff
@@ -714,6 +717,8 @@ public void upload(BucketName bucket, String objectId, InputStream inputStream)
         ensureBucketExists(bucket);
         InitiateMultipartUploadResult multipartUpload =
                 getClient().initiateMultipartUpload(new InitiateMultipartUploadRequest(bucket.getName(), objectId));
+
+        boolean emptyInputStream = false;
```
Comment (mko-sci) [inline src/main/java/sirius/biz/storage/s3/ObjectStore.java:721]: I would rename the variable to: isInputStreamEmpty

Comment (mko-sci) [reply src/main/java/sirius/biz/storage/s3/ObjectStore.java:721]: Or better: describe from which the input stream originates or what the bool is actualy there to check for

**Review by mko-sci — APPROVED** [review]

File: src/main/java/sirius/biz/storage/s3/ObjectStore.java [hunk 2]
```diff
@@ -726,6 +731,8 @@ public void upload(BucketName bucket, String objectId, InputStream inputStream)
                                       eTags.size());
             }
 
+            emptyInputStream = eTags.isEmpty();
+
             getClient().completeMultipartUpload(new CompleteMultipartUploadRequest(bucket.getName(),
```
Comment (andyHa) [inline src/main/java/sirius/biz/storage/s3/ObjectStore.java:736]: could we just abort the empty upload here? has we know for sure that an empty MP upload is pointless anyway? could be use a conventional upload to ensure that the empty object exists?

**Review by andyHa — CHANGES_REQUESTED** [review]

File: src/main/java/sirius/biz/storage/s3/ObjectStore.java (hunk 2 above)
Comment (andyHa) [reply src/main/java/sirius/biz/storage/s3/ObjectStore.java:736]: maybe we could just call

```
getClient().abortMultipartUpload(new AbortMultipartUploadRequest(bucket.getName(),
                                                                             objectId,
                                                                             multipartUpload.getUploadId()));
                                                                             ```
                                                                             
                                                                             followed by s.th. like
```
getClient().putObject(bucket.getName(),
                                  objectId,"");
```

File: src/main/java/sirius/biz/storage/s3/ObjectStore.java (hunk 1 above)
Comment (sabieber) [reply src/main/java/sirius/biz/storage/s3/ObjectStore.java:721]: Why should the name of the boolean start with `is`? I find this a good rule of thumb for a boolean getter method name, but in the actual variable/field name this is unnecessary noise

File: src/main/java/sirius/biz/storage/s3/ObjectStore.java (hunk 2 above)
Comment (idlira) [reply src/main/java/sirius/biz/storage/s3/ObjectStore.java:736]: In the replication event is indeed pointless. But a "normal" upload is the real question: shall we always assume no data as OK? If so, we should than load a 0-byte file. Aborting the transaction means the object id does not exist in the bucket.
Inside ObjectStore, we cannot know that the space is being used for replication (one can even have the idea to chain replications)... 
This is why I decided to keep it non-breaking, preserving the previous behavior and only catch the case when executing a replication.

**Pull request closed without merging**